src/backend/ZipFS.ts

```diff
@@ -626,11 +626,18 @@ export default class ZipFS extends SynchronousFileSystem implements FileSystem {
   }
 
   private static _computeIndexResponsiveTrampoline(data: Buffer, index: FileIndex<CentralDirectory>, cdPtr: number, cdEnd: number, cb: BFSCallback<ZipTOC>, cdEntries: CentralDirectory[], eocd: EndOfCentralDirectory) {
+    let err: ApiError | null = null;
+    let zipTOC: ZipTOC | undefined;
     try {
-      ZipFS._computeIndexResponsive(data, index, cdPtr, cdEnd, cb, cdEntries, eocd);
+      const callbackWrapper: BFSCallback<ZipTOC> = (_err: ApiError, _zipTOC: ZipTOC) => {
+        err = _err;
+        zipTOC = _zipTOC;
+      };
+      ZipFS._computeIndexResponsive(data, index, cdPtr, cdEnd, callbackWrapper, cdEntries, eocd);
```

Comment:
I'm relying on a callback here being synchronous, but that function is not actually always synchronous. If anything it should always be async and we could rely on it being async - in which case I think the callback in try would be okay?

Reply:
I see; yeah, it's not always synchronous. There's a dumb hacky way to get around the issue, but I hesitate to suggest it, as it'll make the code nasty.

I can handle refactoring / fixing this specific issue, so please revert this change that you've made.

Reply:
Sure!